### PR TITLE
fix: `numPositiveBalances` mgmt

### PR DESCRIPTION
**Pull request by tinker-michaelj**
**Description**:
 - Allow `TopicUpdate` to use an empty submit key.
 - In `TokenMintHandler`, ensure the updated treasury account has incremented `numberPositiveBalances`.
 - In `TokenUpdateHandler`, ensure the previous treasury's `numberPositiveBalances` is decremented.

### CODE REVIEW

**Review by Neeharika-Sompalli — DISMISSED** [review]
LGTM!

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenMintHandler.java [hunk 1]
```diff
@@ -214,6 +211,7 @@ private List<Long> mintNonFungible(
 
         // Change the supply on token
         changeSupply(token, treasuryRel, metadataCount, FAIL_INVALID, accountStore, tokenStore, tokenRelStore);
+        treasuryAccount = accountStore.get(treasuryRel.accountIdOrThrow());
```
Comment (mhess-swl) [inline hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenMintHandler.java:214]: Why assign `treasuryAccount` twice?

Comment (Neeharika-Sompalli) [reply hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenMintHandler.java:214]: We are updating `treasuryAccount` in `changeSupply` and were not getting latest account from store again.
This was causing `numPositiveBalances` to be set incorrectly.

**Review by iwsimon — DISMISSED** [review]
LGTM!

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenMintHandler.java (hunk 1 above)
Comment (netopyr) [reply hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenMintHandler.java:214]: We should add a comment. I was confused, too.

Comment (Neeharika-Sompalli) [reply hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenMintHandler.java:214]: Yes, will add now.

Comment (Neeharika-Sompalli) [reply hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenMintHandler.java:215]: Added comments. Thanks @mhess-swl @netopyr

**Review by iwsimon — DISMISSED** [review]
LGTM

**Review by steven-sheehy — COMMENTED** [review]
Shouldn't there be tests that verify these changes? Especially at the Hapi spec level as we may refactor units

**Comment by tinker-michaelj** [comment]
> Shouldn't there be tests that verify these changes? Especially at the Hapi spec level as we may refactor units

Added two EETs [here](https://github.com/hashgraph/hedera-services/pull/12196/files#diff-e66f78f615a9ef8db967bb09893c31958771b5d258f471ac29421e0ffcae1e77R131).

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM !

**Pull request merged**